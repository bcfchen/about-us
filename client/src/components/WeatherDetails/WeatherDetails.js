import React from "react";
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Grow from '@material-ui/core/Grow';
import { propTypes } from "./types";

const WeatherDetails = ({ isLoading, currentHourlyForecast }) => {
    return (
        <Grow in={!isLoading}>
            <Grid container className="weather-details-item-container">
                <Grid item xs>
                    <Typography variant="subheading">Humidity</Typography>
                    <Typography className="humidity-value" variant="subheading">{currentHourlyForecast && currentHourlyForecast.humidity}</Typography>
                </Grid>
                <Grid item xs>
                    <Typography variant="subheading">Wind</Typography>
                    <Typography className="wind-speed-value" variant="subheading">{currentHourlyForecast && currentHourlyForecast.windSpeed}</Typography>
                </Grid>
                <Grid item xs>
                    <Typography variant="subheading">UV Index</Typography>
                    <Typography className="uv-index-value" variant="subheading">{currentHourlyForecast && currentHourlyForecast.uvIndex}</Typography>
                </Grid>
            </Grid>
        </Grow>
    );
};

WeatherDetails.propTypes = propTypes;
export default WeatherDetails;