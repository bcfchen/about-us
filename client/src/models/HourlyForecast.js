import moment from "moment";

class HourlyForecast {
    constructor(properties, location) {
        this.temperature = properties.Temperature.Value;
        this.text = properties.IconPhrase;
        this.dateTime = properties.DateTime;
        this.windSpeed = properties.Wind.Speed.Value + " km/h";
        this.humidity = properties.RelativeHumidity + " %";
        this.uvIndex = properties.UVIndex;
        this.location = location;
    }

    getHoursAndMinutes() {
        return moment(this.dateTime).format("HH:mm A");
    }

    getLocationString() {
        return this.location.getLocationString();
    }
}

export default HourlyForecast;