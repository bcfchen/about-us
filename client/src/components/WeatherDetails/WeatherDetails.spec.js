import React from "react";
import { shallow } from "enzyme";
import WeatherDetails from "./WeatherDetails";
import HourlyForecast from "../../models/HourlyForecast";

describe("<WeatherDetails/>", () => {
    let defaultWeatherProps = {
        Temperature: {
            Value: 15
        },
        Wind: {
            Speed: {
                Value: 6
            }
        },
        UVIndex: 5,
        RelativeHumidity: 50
    };
    let defaultProps = {
        isLoading: true,
        currentHourlyForecast: new HourlyForecast(defaultWeatherProps)
    };
    let component = null;

    let render = (props = defaultProps) => {
        component = shallow(<WeatherDetails {...props} />);
    };

    describe("on page load", () => {
        it("should render correctly", () => {
            render();
            expect(component).toMatchSnapshot();
        });

        describe("if current hourly forecast exists", () => {
            beforeEach(() => {
                render();
            });

            it("should display humidity, wind speed, and UV index", () => {
                expect(component.find(".humidity-value").dive().dive().text()).toEqual("50 %");
                expect(component.find(".wind-speed-value").dive().dive().text()).toEqual("6 km/h");
                expect(component.find(".uv-index-value").dive().dive().text()).toEqual("5");
            });
        });
    });


    describe("if current hourly forecast does not exist", () => {
        beforeEach(() => {
            render({ isLoading: true, currentHourlyForecast: null });
        });

        it("should display humidity, wind speed, and UV index", () => {
            expect(component.find(".humidity-value").dive().dive().text()).toEqual("");
            expect(component.find(".wind-speed-value").dive().dive().text()).toEqual("");
            expect(component.find(".uv-index-value").dive().dive().text()).toEqual("");
        });
    });
});
