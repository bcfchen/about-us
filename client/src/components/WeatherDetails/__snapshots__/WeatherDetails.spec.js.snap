// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDetails/> on page load should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherDetails
    currentHourlyForecast={
        HourlyForecast {
            "dateTime": undefined,
            "humidity": "50 %",
            "isDaylight": undefined,
            "location": undefined,
            "temperature": 15,
            "text": undefined,
            "uvIndex": 5,
            "windSpeed": "6 km/h",
          }
    }
    isLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Grid)
        className="weather-details-item-container"
        container={true}
>
        <WithStyles(Grid)
                item={true}
                xs={true}
        >
                <WithStyles(Typography)
                        variant="subheading"
                >
                        Humidity
                </WithStyles(Typography)>
                <WithStyles(Typography)
                        className="humidity-value"
                        variant="subheading"
                >
                        50 %
                </WithStyles(Typography)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
                item={true}
                xs={true}
        >
                <WithStyles(Typography)
                        variant="subheading"
                >
                        Wind
                </WithStyles(Typography)>
                <WithStyles(Typography)
                        className="wind-speed-value"
                        variant="subheading"
                >
                        6 km/h
                </WithStyles(Typography)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
                item={true}
                xs={true}
        >
                <WithStyles(Typography)
                        variant="subheading"
                >
                        UV Index
                </WithStyles(Typography)>
                <WithStyles(Typography)
                        className="uv-index-value"
                        variant="subheading"
                >
                        5
                </WithStyles(Typography)>
        </WithStyles(Grid)>
</WithStyles(Grid)>,
      "in": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Grid)
            item={true}
            xs={true}
>
            <WithStyles(Typography)
                        variant="subheading"
            >
                        Humidity
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className="humidity-value"
                        variant="subheading"
            >
                        50 %
            </WithStyles(Typography)>
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={true}
>
            <WithStyles(Typography)
                        variant="subheading"
            >
                        Wind
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className="wind-speed-value"
                        variant="subheading"
            >
                        6 km/h
            </WithStyles(Typography)>
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={true}
>
            <WithStyles(Typography)
                        variant="subheading"
            >
                        UV Index
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className="uv-index-value"
                        variant="subheading"
            >
                        5
            </WithStyles(Typography)>
</WithStyles(Grid)>,
        ],
        "className": "weather-details-item-container",
        "container": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="subheading"
>
                Humidity
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className="humidity-value"
                variant="subheading"
>
                50 %
</WithStyles(Typography)>,
            ],
            "item": true,
            "xs": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Humidity",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "Humidity",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "50 %",
                "className": "humidity-value",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "50 %",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="subheading"
>
                Wind
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className="wind-speed-value"
                variant="subheading"
>
                6 km/h
</WithStyles(Typography)>,
            ],
            "item": true,
            "xs": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Wind",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "Wind",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "6 km/h",
                "className": "wind-speed-value",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "6 km/h",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="subheading"
>
                UV Index
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className="uv-index-value"
                variant="subheading"
>
                5
</WithStyles(Typography)>,
            ],
            "item": true,
            "xs": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "UV Index",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": "UV Index",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 5,
                "className": "uv-index-value",
                "variant": "subheading",
              },
              "ref": null,
              "rendered": 5,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Grid)
          className="weather-details-item-container"
          container={true}
>
          <WithStyles(Grid)
                    item={true}
                    xs={true}
          >
                    <WithStyles(Typography)
                              variant="subheading"
                    >
                              Humidity
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              className="humidity-value"
                              variant="subheading"
                    >
                              50 %
                    </WithStyles(Typography)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    item={true}
                    xs={true}
          >
                    <WithStyles(Typography)
                              variant="subheading"
                    >
                              Wind
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              className="wind-speed-value"
                              variant="subheading"
                    >
                              6 km/h
                    </WithStyles(Typography)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    item={true}
                    xs={true}
          >
                    <WithStyles(Typography)
                              variant="subheading"
                    >
                              UV Index
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              className="uv-index-value"
                              variant="subheading"
                    >
                              5
                    </WithStyles(Typography)>
          </WithStyles(Grid)>
</WithStyles(Grid)>,
        "in": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
              xs={true}
>
              <WithStyles(Typography)
                            variant="subheading"
              >
                            Humidity
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            className="humidity-value"
                            variant="subheading"
              >
                            50 %
              </WithStyles(Typography)>
</WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              xs={true}
>
              <WithStyles(Typography)
                            variant="subheading"
              >
                            Wind
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            className="wind-speed-value"
                            variant="subheading"
              >
                            6 km/h
              </WithStyles(Typography)>
</WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              xs={true}
>
              <WithStyles(Typography)
                            variant="subheading"
              >
                            UV Index
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            className="uv-index-value"
                            variant="subheading"
              >
                            5
              </WithStyles(Typography)>
</WithStyles(Grid)>,
          ],
          "className": "weather-details-item-container",
          "container": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  variant="subheading"
>
                  Humidity
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  className="humidity-value"
                  variant="subheading"
>
                  50 %
</WithStyles(Typography)>,
              ],
              "item": true,
              "xs": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Humidity",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": "Humidity",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "50 %",
                  "className": "humidity-value",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": "50 %",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  variant="subheading"
>
                  Wind
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  className="wind-speed-value"
                  variant="subheading"
>
                  6 km/h
</WithStyles(Typography)>,
              ],
              "item": true,
              "xs": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Wind",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": "Wind",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "6 km/h",
                  "className": "wind-speed-value",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": "6 km/h",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  variant="subheading"
>
                  UV Index
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  className="uv-index-value"
                  variant="subheading"
>
                  5
</WithStyles(Typography)>,
              ],
              "item": true,
              "xs": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "UV Index",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": "UV Index",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": 5,
                  "className": "uv-index-value",
                  "variant": "subheading",
                },
                "ref": null,
                "rendered": 5,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
